File: backend/Database/Models/Sprint.cs
namespace SprintTracker.Database.Models
{
    public class Sprint
    {
        public int Id { get; set; }
        public required string Name { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public SprintStatus Status { get; set; } = SprintStatus.Upcoming;
        public List<UserStory> UserStories { get; set; } = new List<UserStory>();
        public string JoinCode { get; set; }
        public List<SprintMember> SprintMembers { get; set; } = new List<SprintMember>();

        public Sprint()
        {
            JoinCode = GenerateJoinCode();
        }

        private static string GenerateJoinCode()
        {
            const string chars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
            var random = new Random();
            return new string(Enumerable.Repeat(chars, 6)
                .Select(s => s[random.Next(s.Length)]).ToArray());
        }
    }
}

File: frontend/src/types/sprint.ts
export const SPRINT_STATUSES = ['Upcoming', 'InProgress', 'Done'] as const;

export type SprintStatus = (typeof SPRINT_STATUSES)[number];

export interface Sprint {
  id: string;
  goal: string;
  startDate: Date;
  endDate: Date;
  status: SprintStatus;
  joinCode?: string;
}

export interface SprintApiResponse {
  id: number | string;
  name?: string;
  startDate: string;
  endDate: string;
  status?: SprintStatus | string;
  joinCode?: string;
}
